src/components/settings/SpendingLimits/NewSpendingLimit/steps/ReviewSpendingLimit.tsx

```diff
@@ -106,8 +108,22 @@ export const ReviewSpendingLimit = ({ data, onSubmit }: Props) => {
     return createNewSpendingLimitTx(data, spendingLimits, chainId, decimals, existingSpendingLimit)
   }, [data, spendingLimits, chainId, decimals, existingSpendingLimit])
 
+  const onNewSpendingLimitTx = (data: null) => {
```

Comment:
General recommendation: don't invent custom fn names, just call it `onFormSubmit` or some such. Easier to follow.